src/components/Header/Header.tsx

```diff
@@ -2,8 +2,13 @@ import React from 'react';
 import { Logo } from '../Logo/Logo';
 import { HeaderBtn } from '../HeaderBtn/HeaderBtn';
 import { UserProfile } from '../UserProfile/UserProfile';
+import { CurrentUserContext } from '../../pages/HomePage/HomePage';
 
 export const Header: React.FC = () => {
+  const currentUser = React.useContext(CurrentUserContext);
```

Comment:
you will store more that only user profile in context. make sence to think about it now. You can provide custom hook to pick up user profile from store

````
const useUserProfile = (): UserPorfile | null => {
  const { user } = useContext(CurrentUserContext);
  return user;
}
````